documentation/source/product-model.rst

```diff
@@ -727,3 +784,295 @@ The ``ProductTaxonomy`` data structure and an instance of a CDM object (`seriali
       "taxonomySource": "ISDA"
     }
   ]
+
+Predefine classifications are available in the model using the AssetClassEnum:
+
+.. code-block:: Haskell
+
+ enum AssetClassEnum:
+     Commodity
+     Credit
+     Equity
+     ForeignExchange
+     InterestRate 
+     MoneyMarket
+	 
+Repurchase Agreements are classified as a MoneyMarket asset but this attribute is not required to qualify a repurchase agreement.
+
+
+Collateral
+^^^^^^^^^^
+
+Collateral on a repo transaction is defined in the AssetPayout object. 
+Security identification is created in the securityInformation attribute 
+which itself is a ProductType allow for the possibility of creating 
+products defined in terms of other products using the same Product model.
+The assetPayoutLeg defines the delivery dates, as relative references to 
+purchase date and repurchase date, and delivery methods (DVP). 
+
+type AssetLeg:
+    settlementDate AdjustableOrRelativeDate (1..1)
+    deliveryMethod DeliveryMethodEnum (1..1)
+	
+	
+When represented in json format, collateral looks like:
+
+Collateral is defined in the assetPayout->securityInformation:
+
+	"securityInformation": {
+		"meta": {
+			"globalKey": "40232adf"
+		},
+		"security": {
+			"productIdentifier": [
+				{
+					"value": {
+						"identifier": {
+							"value": "GB00B24FF097"
+						},
+						"meta": {
+							"globalKey": "1ad6d034"
+						},
+						"source": "ISIN"
+					}],
+			"securityType": "DEBT"
+			}
+		}
+
+
+
+
+
+
+Haircut and Margin
+^^^^^^^^^^^^^^^^^^
+
+Most repo trades include a haircut or margin adjustment to the collateral value that affords the collateral holder a level of risk protection in the case of default and the value of the collateral is lower than the loan value. 
+Haircuts and margin adjustments are set on the collateralProvision under economicTerms->collateral-> collateralProvisions. 
+
+.. code-block:: Haskell
+
+
+type CollateralProvisions:
+
+    collateralType CollateralTypeEnum (1..1)
+        [docReference ICMA GMRA namingConvention "marginType"]
+    eligibleCollateral EligibleCollateralSchedule (0..*) 
+    substitutionProvisions SubstitutionProvisions (0..1)
+
+
+Haircuts in json format appear as:
+
+	"collateralProvisions": {
+		"collateralType": "CASH",
+		"marginPercentage": {
+			"marginPercentage": 0.99
+		}
+	}
+	
+	
+Repo Product in JSON Format
```

Comment:
@tomhealey-icma - including an entire JSON object in the documentation doesn't feel like the best approach to me as this structure could change as the model adjusts, and I don't believe there is any validation to ensure this stays consistent.
Proposed approach would be to include a .json file in the examples project and then have the documentation point to that file.

Reply:
@nigelcobb -Thanks for the suggestion. Is there already a folder for json files and would have an example of how the file linking works in the .rst file.

Reply:
@tomhealey-icma - I agree with Nigel that it would be better if the full object was not in the documentation but would be very useful to include with the CDM distribution :)